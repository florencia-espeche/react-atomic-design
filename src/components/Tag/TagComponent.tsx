import React from 'react'
import { useTheme } from '../../hooks/useTheme'

import { Container } from './TagComponent.styles'
import { TagProps } from './TagComponent.types'

const TagComponent = ({ label, type = 'default' }: TagProps) => {
    const theme = useTheme()

    return (
        <Container theme={theme} type={type}>
            {label}
        </Container>
    )
}

export default TagComponent
