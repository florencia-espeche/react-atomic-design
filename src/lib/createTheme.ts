import { defaultTheme } from '../styles/defaultTheme'
import { Theme } from '../styles/types'

export const createTheme = <T>(theme: Theme & T) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },
        borderRadius: theme.borderRadius || defaultTheme.borderRadius,
        fontTokens: {
            ...defaultTheme.fontTokens,
            ...theme.fontTokens
        },
        fontWeight: theme.fontWeight || defaultTheme.fontWeight,
        name: theme.name || defaultTheme.name
    }
}
