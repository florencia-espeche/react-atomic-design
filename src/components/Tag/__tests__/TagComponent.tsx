import { render } from '../../../lib/test-utils'

import TagComponent from '../TagComponent'

describe('<TagComponent />', () => {
    it('should render the component', () => {
        const { container } = render(<TagComponent label="teste" />)

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render the success version', () => {
        const { container } = render(
            <TagComponent label="teste" type="success" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render the inactive version', () => {
        const { container } = render(
            <TagComponent label="teste" type="inactive" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render the danger version', () => {
        const { container } = render(
            <TagComponent label="teste" type="danger" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should render the warning version', () => {
        const { container } = render(
            <TagComponent label="teste" type="warning" />
        )

        expect(container.firstChild).toMatchSnapshot()
    })
})
