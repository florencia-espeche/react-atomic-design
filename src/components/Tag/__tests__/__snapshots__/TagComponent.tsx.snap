// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagComponent /> should render the component 1`] = `
<div
  class="sc-bdfBQB gTrNY"
  type="default"
>
  teste
</div>
`;

exports[`<TagComponent /> should render the danger version 1`] = `
<div
  class="sc-bdfBQB goxGPs"
  type="danger"
>
  teste
</div>
`;

exports[`<TagComponent /> should render the inactive version 1`] = `
<div
  class="sc-bdfBQB eobTVd"
  type="inactive"
>
  teste
</div>
`;

exports[`<TagComponent /> should render the success version 1`] = `
<div
  class="sc-bdfBQB njdSo"
  type="success"
>
  teste
</div>
`;

exports[`<TagComponent /> should render the warning version 1`] = `
<div
  class="sc-bdfBQB gCVwta"
  type="warning"
>
  teste
</div>
`;
